client/src/periodUtils.js

```diff
@@ -159,3 +159,41 @@ PeriodsTableHeader.propTypes = {
     PeriodsConfigPropType
   ])
 }
+export const ONE_PERIOD_WIDTH_LIMIT = 560
+
+const SCREEN_SIZES = {
+  largeLowLimit: 1000,
+  midLowLimit: 600
+  // further lower is small, no need for a limit
+}
+
+const SCREEN_SIZE_TO_PERIOD_NUMBER = {
+  large: {
+    isMatch: width => width >= SCREEN_SIZES.largeLowLimit,
+    no: 3
+  },
+  mid: {
+    isMatch: width =>
+      width > SCREEN_SIZES.midLowLimit && width < SCREEN_SIZES.largeLowLimit,
+    no: 2
+  },
+  small: {
+    isMatch: width => width <= SCREEN_SIZES.midLowLimit,
+    no: 1
+  }
+}
+
+function getPeriodNumberForScreen(width) {
+  for (const size in SCREEN_SIZE_TO_PERIOD_NUMBER) {
+    if (SCREEN_SIZE_TO_PERIOD_NUMBER[size].isMatch(width)) {
+      return SCREEN_SIZE_TO_PERIOD_NUMBER[size].no
+    }
+  }
+}
+
+export const useLessNumberOfPeriodsOnSmallScreens = setNumberOfPeriods => {
```

Comment:
I'd rename this function to `useFewerPeriodsOnSmallScreens` ("less periods" is grammatically incorrect, and on top of that it looks like a "useless" function :wink: ).

Reply:
Ah, yes. Actually, i should probably rename something like "useResponsiveNumberOfPeriods". Hiding the "Desktop-first" bias :smile: